tests/Behat/features/element-editor.feature

```diff
@@ -11,7 +11,7 @@ Feature: View types of elements in an area on a page
     And the "page" "Blocks Page" has a "Bob's Block" content element with "Some content II" content
 
   Scenario Outline: I can see the title and summary of each element
-    Given I am logged in with "<group>" permissions
+    Given I am logged in with "AUTHOR" permissions
```

Comment:
`And a "group" "AUTHOR group" has permissions "Access to 'Pages' section"`

How I see we create special group with restricted permissions. I can rename AUTHOR to EDITOR to more consistent in naming.

Reply:
Ahh, okay. So it says "permissions" but in this case it actually means the _group_.

We probably want to change that to avoid confusion. For these where you are logging in with a _group_, can you please do the following:

1. In [silverstripe/silverstripe-behat-extension](https://github.com/silverstripe/silverstripe-behat-extension/)'s `LoginContext` class, add a new `public function iAmLoggedInWithGroup($group)` method:
    ```php
    /**
     * Creates a member in a specific group.
     * Example: Given I am logged in with the "Content Author" group
     *
     * @Given /^I am logged in with the "([^"]*)" group$/
     * @param string $permCode
     */
    public function iAmLoggedInWithGroup($group)
    {
        $email = "{$group}@example.org";
        $password = 'Secret!123';
        // Get or create group
        $group = Group::get()->filter('Title', $group)->first();
        if (!$group) {
            throw new LogicException("Group $group does not exist");
        }
        // Use the logic from generateMemberWithPermission to add a new member to the group
        // You may want to do some refactoring to avoid duplicated code
        $this->stepILogInWith($email, $password);
    }
    ```
1. Instead of naming the group "AUTHOR group" name it something descriptive like "elemental block editor"
1. Use the new `And I am logged in with the "elemental block editor" group` syntax

Reply:
Do you suggest it for all test cases where we use this `Given I am logged in with "*" permissions` approach?

Reply:
Only if it's actually referring to a group.
E.g. `Given I am logged in with "ADMIN" permissions` is actually referring to a permission, so that's fine and should be left as is.

Reply:
If deal only in sentence meaning, I can create new method only for this
```    /**
     * Creates and login as a member of a group with the correct permissions.
     * Example: Given I am logged in as a memeber of "ADMIN" group
     *
     * @Given /^I am logged in as memeber of "([^"]*)" group$/
     * @param string $permCode
     */
    public function iAmLoggedInAsMemeberOfGroup($permCode)
    {
        $this->iAmLoggedInWithPermissions($permCode);
    }
    ```

Reply:
That solution would require the group to be prefixed with " group" and then you would have to reference it without that - e.g:
```
Given the "group" "elemental block editor group" has permissions "Access to 'Pages' section" and "ELEMENTAL_DATAOBJECT_EDIT"
And I am logged in with the "elemental block editor" group
```
That's not ideal. It's a bit confusing that you need to create the group with the word "group" at the end, but you don't actually use the word "group" a the end of the group name when logging in with it.

Reply:
Actually, it's more confusing, since a user can log in as member, not a group or a role. 
Or a user can log in with permission of the group, not with group.

Reply:
I'm not sure I follow what you mean... in either case I think the logic in the method snippet I pasted above is correct. I'll leave it up to you to decide the exact syntax for the sentence structure - so long as it's clear that you're intended to log into a user which is in a specific group which you have already set up (as opposed to logging in with specific permissions).